sklearn/tests/test_multiclass.py

```diff
@@ -428,6 +428,41 @@ def test_ovo_fit_predict():
     assert_equal(len(ovo.estimators_), n_classes * (n_classes - 1) / 2)
 
 
+def test_ovo_decision_function():
+    n_samples = iris.data.shape[0]
+
+    ovo_clf = OneVsOneClassifier(LinearSVC(random_state=0))
+    ovo_clf.fit(iris.data, iris.target)
+    decisions = ovo_clf.decision_function(iris.data)
+
+    assert_equal(decisions.shape, (n_samples, n_classes))
+    assert_array_equal(decisions.argmax(axis=1), ovo_clf.predict(iris.data))
+
+    # Compute the votes and scores
+    votes = np.zeros((n_samples, n_classes))
+    scores = np.zeros((n_samples, n_classes))
+
+    k = 0
+    for i in range(n_classes):
+        for j in range(i + 1, n_classes):
+            pred = ovo_clf.estimators_[k].predict(iris.data)
+            score = ovo_clf.estimators_[k].decision_function(iris.data)
+            scores[:, i] -= score
+            scores[:, j] += score
+            votes[pred == 0, i] += 1
+            votes[pred == 1, j] += 1
+            k += 1
+
+    # Extract votes and verify
+    assert_array_equal(votes, np.round(decisions))
+
+    # Extract the scores and unscale them.
+    scores_extracted = decisions - votes
+    scores_extracted *= max(abs(scores.max()), abs(scores.min())) / 0.4
+
+    assert_almost_equal(scores_extracted, scores)
```

Comment:
I think we don't need to test that are we are not interested in the precise formula to compute `decisions` from pair-wise `votes` and `scores`. Instead I think it would be interesting to check that `decisions` has no ties (fix the RNG seed to do so) while `votes` has many.